app/controllers/user_journey_controller.rb

```diff
@@ -87,4 +82,35 @@ def confirm
   def find_certificate
     @certificate = Certificate.find_by_id(params[:certificate_id])
   end
+
+  def merge_errors(primary, *objects)
+    objects.compact.each { |object| primary.errors.merge!(object.errors) }
+    primary.errors.full_messages.join(', ')
+  end
+
+  def error_class(type)
+    if errors_present?
+      if type == 'file' && params['upload-certificate'] == 'file'
```

Comment:
do we need to check the params too?
```suggestion
      if type == 'file'
```

Reply:
Unfortunately not. Without both the when an error occurs both radio button values will be highlighted red

Reply:
oh I see now! sorry!